common/djangoapps/util/model_utils.py

```diff
@@ -143,3 +148,47 @@ def _get_truncated_setting_value(value, max_length=None):
         return value[0:max_length], True
     else:
         return value, False
+
+
+# Taken from Django 1.8 source code because it's not supported in 1.4
+def slugify(value):
+    """Converts value into a string suitable for readable URLs.
+
+    Converts to ASCII. Converts spaces to hyphens. Removes characters that
+    aren't alphanumerics, underscores, or hyphens. Converts to lowercase.
+    Also strips leading and trailing whitespace.
+
+    Args:
+        value (string): String to slugify.
+    """
+    value = force_unicode(value)
+    value = unicodedata.normalize('NFKD', value).encode('ascii', 'ignore').decode('ascii')
+    value = re.sub(r'[^\w\s-]', '', value).strip().lower()
+    return mark_safe(re.sub(r'[-\s]+', '-', value))
+
+
+def generate_unique_readable_id(name, queryset, lookup_field):
```

Comment:
Actually, since this requires a queryset, I'm OK with it not having an independent unit test (mocking out the queryset would be a pain and not that useful).